File: workflow/plugins/python/var/var_set/var_set.py
"""Workflow plugin: set variable in workflow store."""

from ...base import NodeExecutor


class VarSet(NodeExecutor):
    """Set a variable value in the workflow store."""

    node_type = "var.set"
    category = "var"
    description = "Set variable in workflow store"

    def execute(self, inputs, runtime=None):
        """Set variable value in workflow store."""
        key = inputs.get("key")
        value = inputs.get("value")

        if key is None:
            return {"result": None, "key": None, "error": "key is required"}

        runtime.store[key] = value

        return {"result": value, "key": key}

File: workflow/plugins/go/var/var_set/var_set.go
// Package var_set provides a workflow plugin for setting workflow variables.
package var_set

// VarSet implements the NodeExecutor interface for setting workflow variables.
type VarSet struct {
	NodeType    string
	Category    string
	Description string
}

// NewVarSet creates a new VarSet instance.
func NewVarSet() *VarSet {
	return &VarSet{
		NodeType:    "var.set",
		Category:    "var",
		Description: "Set a variable in the workflow store",
	}
}

// Runtime interface for accessing workflow store.
type Runtime interface {
	GetStore() map[string]interface{}
}

// Execute runs the plugin logic.
// Stores a variable in the workflow store.
func (p *VarSet) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	key, ok := inputs["key"].(string)
	if !ok {
		return map[string]interface{}{"success": false, "error": "key is required"}
	}

	value := inputs["value"]

	// Try to access the runtime store
	var store map[string]interface{}
	if r, ok := runtime.(Runtime); ok {
		store = r.GetStore()
	} else if r, ok := runtime.(map[string]interface{}); ok {
		if s, ok := r["Store"].(map[string]interface{}); ok {
			store = s
		}
	}

	if store == nil {
		return map[string]interface{}{"success": false, "error": "runtime store not available"}
	}

	store[key] = value

	return map[string]interface{}{"success": true, "key": key}
}
